help_to_heat/frontdoor/views.py

```diff
@@ -641,6 +649,16 @@ def save_post_data(self, data, session_id, page_name):
             logger.exception(f"An error occurred: {e}")
             data[address_choice_field] = address_choice_field_epc_api_fail
 
+        will_use_os_api = (
+            data[address_choice_field] == address_choice_field_epc_api_fail
+            or country == country_field_scotland
+            and data[address_choice_field] == address_choice_field_write_address
+        )
```

Comment:
Are we now re-implementing logic from above (which could be equally brittle)? This is why I'd consider returning early in the case(s) where we don't want to do it.